### PR TITLE
Add first version of .NET SDK for Policy Insights RP

### PR DESCRIPTION
**Pull request by bulentelmaci**
## Description
First version of .NET SDK generated based off of our new swagger spec. The PR merged in the azure-rest-api-specs repository is https://github.com/Azure/azure-rest-api-specs/pull/2523 (we later reorganized the files with PR https://github.com/Azure/azure-rest-api-specs/pull/2599). The specs are at https://github.com/Azure/azure-rest-api-specs/tree/master/specification/policyinsights/resource-manager.

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/PolicyInsights/Management/Management.PolicyInsights/Microsoft.Azure.Management.PolicyInsights.csproj
Comment (shahabhijeet) [inline src/SDKs/PolicyInsights/Management/Management.PolicyInsights/Microsoft.Azure.Management.PolicyInsights.csproj]: @bulentelmaci if this is your v1 and you are planning to go preview, I would recommend you go as 0.9.0-preview.
Because when you are ready to go stable (non preview) you can then mark it as 1.0.0 and that will make your customers take notice on your first stable version of your SDK.
All V1 products go out as 0.9.0-preview

File: src/SDKs/PolicyInsights/Management/PolicyInsights.Tests/PolicyInsights.Tests.csproj
Comment (shahabhijeet) [inline src/SDKs/PolicyInsights/Management/PolicyInsights.Tests/PolicyInsights.Tests.csproj]: @bulentelmaci remove commented code

File: src/SDKs/PolicyInsights/Management/PolicyInsights.Tests/PolicyInsights.Tests.csproj
Comment (shahabhijeet) [inline src/SDKs/PolicyInsights/Management/PolicyInsights.Tests/PolicyInsights.Tests.csproj]: remove commented code

File: src/SDKs/PolicyInsights/Management/Management.PolicyInsights/Microsoft.Azure.Management.PolicyInsights.csproj
Comment (shahabhijeet) [inline src/SDKs/PolicyInsights/Management/Management.PolicyInsights/Microsoft.Azure.Management.PolicyInsights.csproj]: @bulentelmaci package tags should be single word, adding space makes it two tags.
I am trying to get rid of tags that have spaces in it. you can put as many tags you want as long as those tags can be used for customers to search for your nuget package.

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
use generate.cmd to generate your sdk.
this will generate the meta data file that is needed.
Once you generate using generate.cmd, you will then have to build your SDK at least once using
msbuild build.proj /t:build /p:Scope=SDKs\PolicyInsights

File: src/SDKs/PolicyInsights/Management/PolicyInsights.Tests/PolicyInsightsTests.cs
Comment (shahabhijeet) [inline src/SDKs/PolicyInsights/Management/PolicyInsights.Tests/PolicyInsightsTests.cs]: @bulentelmaci this will not work if I ever try to record your tests? Any reasons you are hardcoding these paths?

File: src/SDKs/PolicyInsights/Management/PolicyInsights.Tests/PolicyInsightsTests.cs
Comment (shahabhijeet) [inline src/SDKs/PolicyInsights/Management/PolicyInsights.Tests/PolicyInsightsTests.cs]: @bulentelmaci you never want to set anything in Playback because you are not authenticating anything. All you are doing is replaying your recorded files.
Also you don't want to set anything in code during record mode. What you should be doing is, save the connection string for your recording in a keyVault. Please take away all your dependencies hard coded

File: src/SDKs/PolicyInsights/Management/PolicyInsights.Tests/PolicyInsightsTests.cs
Comment (shahabhijeet) [inline src/SDKs/PolicyInsights/Management/PolicyInsights.Tests/PolicyInsightsTests.cs]: @bulentelmaci does this mean I should be having a pre-created assets e.g. ManagementGroup called AzGovTest1? in order to record this test?

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

File: src/SDKs/PolicyInsights/Management/Management.PolicyInsights/Microsoft.Azure.Management.PolicyInsights.csproj
Comment (bulentelmaci) [reply src/SDKs/PolicyInsights/Management/Management.PolicyInsights/Microsoft.Azure.Management.PolicyInsights.csproj]: Will do

File: src/SDKs/PolicyInsights/Management/PolicyInsights.Tests/PolicyInsights.Tests.csproj
Comment (bulentelmaci) [reply src/SDKs/PolicyInsights/Management/PolicyInsights.Tests/PolicyInsights.Tests.csproj]: Will do

File: src/SDKs/PolicyInsights/Management/PolicyInsights.Tests/PolicyInsights.Tests.csproj
Comment (bulentelmaci) [reply src/SDKs/PolicyInsights/Management/PolicyInsights.Tests/PolicyInsights.Tests.csproj]: Will do

File: src/SDKs/PolicyInsights/Management/Management.PolicyInsights/Microsoft.Azure.Management.PolicyInsights.csproj
Comment (bulentelmaci) [reply src/SDKs/PolicyInsights/Management/Management.PolicyInsights/Microsoft.Azure.Management.PolicyInsights.csproj]: ok, i'll change those tags then

File: src/SDKs/PolicyInsights/Management/PolicyInsights.Tests/PolicyInsightsTests.cs
Comment (bulentelmaci) [reply src/SDKs/PolicyInsights/Management/PolicyInsights.Tests/PolicyInsightsTests.cs]: I needed to use interactive login to record sessions which means having to use fill .net SDK. With full .net SDK, recording is broken unless you override recordsdirectory with a new one right after context is started because the recordsdirectory set is invalid (conflicts with the DLL name).

File: src/SDKs/PolicyInsights/Management/PolicyInsights.Tests/PolicyInsightsTests.cs
Comment (bulentelmaci) [reply src/SDKs/PolicyInsights/Management/PolicyInsights.Tests/PolicyInsightsTests.cs]: Without the subscription ID, playback mode failed for me, hence I did this. I can remove all these hardcoded stuff for sure, but it will make life painful again next time we want to record sessions (will need an app id, and using interactive login wont work, etc.).

File: src/SDKs/PolicyInsights/Management/PolicyInsights.Tests/PolicyInsightsTests.cs
Comment (bulentelmaci) [reply src/SDKs/PolicyInsights/Management/PolicyInsights.Tests/PolicyInsightsTests.cs]: Of course. This is an extension RP that relies on real resources to be able to do anything. Management groups, subscription Ids, resource groups, resources, policies, etc.

**Comment by bulentelmaci** [comment]
@shahabhijeet Made all requested changes

**Review by shahabhijeet — APPROVED** [review]

**Comment by shahabhijeet** [comment]
@bulentelmaci this is good to go once the CI succeeds

**Comment by bulentelmaci** [comment]
@shahabhijeet CI seems to have succeeded. Do we need to do anything for the nuget package to be published properly after the merge?

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**
